sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -0,0 +1,412 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.implementation;
+
+import com.azure.core.credentials.AccessToken;
+import com.azure.core.http.ProxyOptions;
+import com.azure.core.implementation.serializer.SerializerAdapter;
+import com.azure.core.implementation.serializer.SerializerEncoding;
+import com.azure.core.implementation.serializer.jackson.JacksonAdapter;
+import com.azure.core.implementation.util.ScopeUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.identity.DeviceCodeChallenge;
+import com.azure.identity.implementation.util.CertificateUtil;
+import com.microsoft.aad.msal4j.AuthorizationCodeParameters;
+import com.microsoft.aad.msal4j.ClientCredentialFactory;
+import com.microsoft.aad.msal4j.ClientCredentialParameters;
+import com.microsoft.aad.msal4j.ConfidentialClientApplication;
+import com.microsoft.aad.msal4j.DeviceCodeFlowParameters;
+import com.microsoft.aad.msal4j.PublicClientApplication;
+import com.microsoft.aad.msal4j.SilentParameters;
+import com.microsoft.aad.msal4j.UserNamePasswordParameters;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.awt.Desktop;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.Proxy;
+import java.net.Proxy.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Scanner;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+/**
+ * The identity client that contains APIs to retrieve access tokens
+ * from various configurations.
+ */
+public class IdentityClient {
```

Comment:
This class should not be public

Reply:
This is in the implementation package - so credentials outside the package cannot see it unless it's public. But since it's in implementation package, no javadocs will be generated for it and Java devs know they shouldn't use it.